### ISSUE TITLE
static-this triggered for this function parameter

### ISSUE THREAD

**Issue by MaceWindu** [palantir/tslint#4739]
### Bug Report

- __TSLint version__: 5.16.0
- __TypeScript version__:  3.4.5
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
export class Fail {
        public static method(): unknown {
            const test = function (this: string) {
                return this === 'that'; // here
            };

            return test;
        }
    }
```

with `tslint.json` configuration:

```json
{
"rules":{
"static-this": true
}
}
```

#### Actual behavior

```
static-this           Use the parent class name instead of `this` when in a `static` context.
```

#### Expected behavior
no linting errors

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
